src/main/java/edu/kis/powp/jobs2d/drivers/decorators/CommandRecorderDriverDecorator.java

```diff
@@ -0,0 +1,27 @@
+package edu.kis.powp.jobs2d.drivers.decorators;
+
+import edu.kis.powp.jobs2d.Job2dDriver;
+import edu.kis.powp.jobs2d.command.OperateToCommand;
+import edu.kis.powp.jobs2d.command.SetPositionCommand;
+import edu.kis.powp.jobs2d.features.RecordingMacro;
+
+public class CommandRecorderDriverDecorator extends BaseJob2dDriverDecorator {
+
+    private RecordingMacro recordingMacro = RecordingMacro.getInstance();
```

Comment:
You should apply dependency injection to avoid fixed coupling with a single instance.